code/game/objects/structures/stool_bed_chair_nest/bed.dm

```diff
@@ -576,13 +576,126 @@
 	held_item = /obj/item/roller/hover
 	has_iv_light = FALSE
 
+	/**
+	 * The color of light the hoverbed emits
+	 */
+	var/hover_color = LIGHT_COLOR_CYAN
+	/**
+	 * The cell powering the hoverbed
+	 */
+	var/obj/item/cell/cell
+	/**
+	 * The power drained from the bed every tick
+	 */
+	var/power_draw = 5
+	/**
+	 * If the hovermode is active
+	 */
+	var/hover = TRUE
+	/**
+	 * If the power cell is accessible or not
+	 */
+	var/cell_open = FALSE
+	/**
+	 * If we have warned about low power recently
+	 */
+	var/warned = FALSE
+
 /obj/structure/bed/roller/hover/Initialize()
 	.=..()
-	set_light(2,1,LIGHT_COLOR_CYAN)
+	set_light(2,1,hover_color)
+	if(!cell || !cell.charge)
+		hover = FALSE
+	START_PROCESSING(SSprocessing, src)
+
+/obj/structure/bed/roller/hover/Destroy()
+	QDEL_NULL(cell)
+	return ..()
+
+// Same as parent, but doesn't kill processing (We need that for power drain)
+/obj/structure/bed/roller/hover/detach_iv(mob/living/carbon/human/target, mob/user)
+	user.visible_message(SPAN_NOTICE("<b>[user]</b> takes [target] off the IV on \the [src]."), SPAN_NOTICE("You take the IV off \the [target]."))
+	iv_attached = FALSE
+	update_icon()
+
+/obj/structure/bed/roller/hover/process()
```

Comment:
You need to respect the `ticks_per_second` parameter that it gives you, and use it for anything that works on time